wit/deps/spin-postgres@4.0.0/postgres.wit

```diff
@@ -0,0 +1,145 @@
+package spin:postgres@4.0.0;
+
+interface postgres {
+  /// Errors related to interacting with a database.
+  variant error {
+      connection-failed(string),
+      bad-parameter(string),
+      query-failed(string),
+      value-conversion-failed(string),
+      other(string)
+  }
+
+  /// Data types for a database column
+  enum db-data-type {
+      boolean,
+      int8,
+      int16,
+      int32,
+      int64,
+      floating32,
+      floating64,
+      str,
+      binary,
+      date,
+      time,
+      datetime,
+      timestamp,
+      uuid,
+      jsonb,
+      decimal,
+      range-int32,
+      range-int64,
+      range-decimal,
+      array-int32,
+      array-int64,
+      array-decimal,
+      array-str,
+      interval,
+      other,
+  }
+
+  /// Database values
+  variant db-value {
+      boolean(bool),
+      int8(s8),
+      int16(s16),
+      int32(s32),
+      int64(s64),
+      floating32(f32),
+      floating64(f64),
+      str(string),
+      binary(list<u8>),
+      date(tuple<s32, u8, u8>), // (year, month, day)
+      time(tuple<u8, u8, u8, u32>), // (hour, minute, second, nanosecond)
+      /// Date-time types are always treated as UTC (without timezone info).
+      /// The instant is represented as a (year, month, day, hour, minute, second, nanosecond) tuple.
+      datetime(tuple<s32, u8, u8, u8, u8, u8, u32>),
+      /// Unix timestamp (seconds since epoch)
+      timestamp(s64),
+      uuid(string),
+      jsonb(list<u8>),
+      decimal(string), // I admit defeat. Base 10
+      range-int32(tuple<option<tuple<s32, range-bound-kind>>, option<tuple<s32, range-bound-kind>>>),
+      range-int64(tuple<option<tuple<s64, range-bound-kind>>, option<tuple<s64, range-bound-kind>>>),
+      range-decimal(tuple<option<tuple<string, range-bound-kind>>, option<tuple<string, range-bound-kind>>>),
+      array-int32(list<option<s32>>),
+      array-int64(list<option<s64>>),
+      array-decimal(list<option<string>>),
+      array-str(list<option<string>>),
+      interval(interval),
+      db-null,
+      unsupported,
```

Comment:
Would it make sense to have a e.g. `raw(list<u8>)`?

Reply:
Could you say more about what you're envisaging?

Reply:
Many postgres drivers provide some mechanism to extend their builtin types by converting to/from raw postgres values: 
https://docs.rs/tokio-postgres/latest/tokio_postgres/types/trait.FromSql.html
https://pkg.go.dev/github.com/jackc/pgx/v4#hdr-Raw_Bytes_Mapping
https://www.psycopg.org/docs/advanced.html#type-casting-from-sql-to-python

Reply:
I guess `raw` would effectively be "other".

Reply:
Gotcha.  Intercepting that may require us a bit more FromSql magic, I am not sure - we access values via `get`, which expects to return into a FromSql implementation that matches the column type, but maybe it will let us return anything into a `Vec<u8>`, or otherwise I can create a fake FromSql that accepts all types and just captures the raw bytes.  I will take a look!

Reply:
Your lightest whim is my command (subject to applicable law and terms of service).  I provisionally put the type name rather than the OID but if you have opinions then bring 'em on, I am utterly naive about this stuff.

Reply:
I don't really know how these are used in practice either, I'm just trying to save you from spending the rest of your days adding support for an neverending list of increasingly obscure data types. 😅

Reply:
Indeed - this has me thinking "what if I rewrote the whole interface to just tunnel EVERYTHING as blobs and let the guest language use whatever decoder is idiomatic" so your plan to free me from this rabbit hole may have backfired...

(Fear not.  I do not intend to do this as part of this PR.  Nor without a _lot_ more research into the factoring of guest language DB libraries.)